code/game/objects/items/weapons/tools/mods/_upgrades.dm

```diff
@@ -333,6 +333,11 @@
 	if(weapon_upgrades[GUN_UPGRADE_BAYONET])
 		G.attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
 		G.sharp = TRUE
+	if(weapon_upgrades[GUN_UPGRADE_FLASHLIGHT])
+		G.flashlight_attachment = TRUE
+		G.verbs += /obj/item/gun/proc/toggle_light
+	else
+		G.verbs -= /obj/item/gun/proc/toggle_light
```

Comment:
This removes the flashlight ability when a mod is added after the flashlight.

Reply:
Oh.